### PR TITLE
rename panic_if_ intrinsics to assert_

### PR DESCRIPTION
**Pull request by RalfJung**
[Suggested by @eddyb](https://github.com/rust-lang/miri/issues/1222#issuecomment-598087523)

### CODE REVIEW

File: src/librustc_codegen_ssa/mir/block.rs [hunk 1]
```diff
@@ -460,9 +460,9 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         };
         let panic_intrinsic = intrinsic.and_then(|i| match i {
             // FIXME: Move to symbols instead of strings.
-            "panic_if_uninhabited" => Some(PanicIntrinsic::IfUninhabited),
-            "panic_if_zero_invalid" => Some(PanicIntrinsic::IfZeroInvalid),
-            "panic_if_any_invalid" => Some(PanicIntrinsic::IfAnyInvalid),
+            "assert_inhabited" => Some(PanicIntrinsic::IfUninhabited),
+            "assert_zero_valid" => Some(PanicIntrinsic::IfZeroInvalid),
+            "assert_uninit_valid" => Some(PanicIntrinsic::IfAnyInvalid),
```
Comment (eddyb) [inline src/librustc_codegen_ssa/mir/block.rs]: The enum could still use some renaming.

**Comment by RalfJung** [comment]
Renamed enum.
@bors r=Centril,eddyb

**Review by eddyb — APPROVED** [review]

**Comment by pietroalbini** [comment]
Had to cancel the CI build to get the whole CI unstuck, sorry for the trouble.

**Pull request merged**